### PR TITLE
add transformer to fix the movie titles

### PR DESCRIPTION
**Pull request by elip06**

### CODE REVIEW

File: sparkify/transformers/fix_movie_titles/movie_titles.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from pyspark.sql import DataFrame
+
+from .movie_title_helpers import fixTitle
+from ..base_transformer import BaseTransformer
+from pyspark.sql.functions import udf
```
Comment (RicoFio) [inline sparkify/transformers/fix_movie_titles/movie_titles.py:5]:
```suggestion
from pyspark.sql import DataFrame
from pyspark.sql.functions

from .movie_title_helpers import fixTitle
from ..base_transformer import BaseTransformer
import udf
```

**Review by RicoFio — APPROVED** [review]
Just a little comment but looks good otherwise

File: sparkify/transformers/fix_movie_titles/movie_title_helpers.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+def fixTitle(title):
+    # remove blankspace before and after title + lowercase
+    fixedTitle = title.strip().lower()
+    # replace weird letters
+    fixedTitle = unidecode(fixedTitle)
+    # remove control characters
+    fixedTitle = "".join(ch for ch in fixedTitle if unicodedata.category(ch)[0]!="C")
+    # remove punctuation
+    fixedTitle = fixedTitle.translate(str.maketrans('', '', string.punctuation))
+    return fixedTitle
```
Comment (DarkCyTic) [inline sparkify/transformers/fix_movie_titles/movie_title_helpers.py:14]:
```suggestion
    return fixedTitle

```

File: sparkify/transformers/fix_movie_titles/movie_titles.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    def _transform(self, dataset: DataFrame):
+
+        @udf
+        def t_function(primaryTitle: str):
+            fixedTitle = fixTitle(primaryTitle)
+            return fixedTitle
+
+        dataset = dataset.withColumn("primaryTitle",t_function(dataset.primaryTitle))
+
+        return dataset
```
Comment (DarkCyTic) [inline sparkify/transformers/fix_movie_titles/movie_titles.py:22]:
```suggestion
        return dataset

```

File: sparkify/transformers/fix_movie_titles/movie_titles.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    SCHEMA = BASE_DATA_SCHEMA
+
+    def _transform(self, dataset: DataFrame):
+
+        @udf
+        def t_function(primaryTitle: str):
+            fixedTitle = fixTitle(primaryTitle)
+            return fixedTitle
+
+        dataset = dataset.withColumn("primaryTitle",t_function(dataset.primaryTitle))
```
Comment (DarkCyTic) [inline sparkify/transformers/fix_movie_titles/movie_titles.py:20]:
```suggestion
        dataset = dataset.withColumn("primaryTitle", t_function(dataset.primaryTitle))
```

**Review by DarkCyTic — APPROVED** [review]
LGTM 🚀 ! Can't wait to re-run the Wikidata searcher with the corrected titles! (small formatting comments)

**Pull request merged**